addons/bus/tests/cleanup.py

```diff
@@ -0,0 +1,40 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from unittest.mock import patch
+from odoo.tests import HttpCase
+from ..websocket import Websocket, ConnectionState, CloseCode
+
+
+def cleanup_websockets_after_test():
+    """
+    Close all the websocket instances that were opened during the test. This
+    method should be called after each test to ensure no websocket is left open.
+    """
+    for websocket in Websocket._instances:
+        if websocket.in_test_mode and websocket.state is ConnectionState.OPEN:
```

Comment:
This check doesn't look really usefull, mu main concern is that `in_test_mode` is actually set  by the mocked_init and this code will actually fail if the websocket is not in test mode, this should use hasattr (if really necessary)

Test usually don't work well if the database is serving requests at the same time, since the request of an httpcase would reach a test cursor. I wouldn't manage this case.